tests/framework/stats/core.py

```diff
@@ -65,11 +67,19 @@ def run_exercise(self, fail_fast=False) -> Result:
         for tag, pipe in self._pipes.items():
             for iteration in range(iterations):
                 raw_data = pipe.producer.produce()
-                if isinstance(raw_data, types.GeneratorType):
-                    for data in raw_data:
-                        pipe.consumer.ingest(iteration, data)
-                else:
-                    pipe.consumer.ingest(iteration, raw_data)
+                if not isinstance(raw_data, types.GeneratorType):
+                    raw_data = [raw_data]
+                for data in raw_data:
+                    raws = pipe.consumer.ingest(iteration, data)
+                    if raws is not None:
+                        dimensions = self.custom.copy()
+                        dimensions["test"] = self.metrics_test
+                        self.metrics.set_dimensions(dimensions)
+                        for name, val, unit in raws:
+                            self.metrics.put_metric(name, val, unit)
```

Comment:
Since these are emitted at a high rate, we should set `storage_resolution = StorageResolution.HIGH`

Reply:
Hmm I chose the numbers in the latency test to have around 1 datapoint per minute. It shouldn't hurt however.

Reply:
huh, but the test is setup to ping 30 times with an interval of 0.2s between pings, so there'd be 5 data points per seconds (which is too much even for high resolution metrics, but oh well)

Reply:
Oh right, so it is 1 datapoint for each iteration, every 6 seconds, sorry. 

I agree it may not work for every test. If they cannot be split into iterations then it doesn't make sense for that test. In that case we can change the data format and return an optional timestamp. If the timestamp is present, we backdate, otherwise we use the regular timestamps.

Reply:
But yeah we can add that if we find this kind of data valuable